projects/angular/src/data/tree-view/models/tree-node.model.ts

```diff
@@ -46,21 +47,32 @@ export abstract class TreeNodeModel<T> {
     this.selected.complete();
   }
 
+  parentSetSelected(state: ClrSelectedState, propagateUp: boolean, propagateDown: boolean) {
+    if (this.disabled) {
+      return;
+    }
+    this.setSelected(state, propagateUp, propagateDown);
+  }
+
   // Propagate by default when eager, don't propagate in the lazy-loaded tree.
   setSelected(state: ClrSelectedState, propagateUp: boolean, propagateDown: boolean) {
     if (state === this.selected.value) {
       return;
     }
     this.selected.next(state);
     if (propagateDown && state !== ClrSelectedState.INDETERMINATE && this.children) {
-      this.children.forEach(child => child.setSelected(state, false, true));
+      this.children.forEach(child => child.parentSetSelected(state, false, true));
```

Comment:
```suggestion
      this.children.forEach(child => {
        if (!child.disabled) {
          child.setSelected(state, false, true);
        }
      });
```

I would remove the `parentSetSelected` method. The name is confusing, and I don't think the method is needed.

Reply:
Why doesn't `setSelected` check for disabled? I feel like it should, but I have not tested it. If `setSelected` checked `disabled` like the other methods, I don't think this line would need any changes at all.

Reply:
Function `parentSetSelected` is removed in commit  6846f60.

`setSelected` function is 'internal' for `[clrSelected]` directive. Selection through it should not be blocked, but only through user input.
TLDR is node with `[clrSelected]=true` and `[clrDisabled]=true` will not be selected.